documentation/src/docs/streaming.md

```diff
@@ -0,0 +1,194 @@
+   - A `CheckPoint` event will be sent by Pravega as part of the data stream flow and on receiving the event, the `FlinkPravegaReader` will initiate [`triggerCheckpoint`](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/checkpoint/ExternallyInducedSource.java#L73) request to effectively let Flink continue and complete the checkpoint process.
+
+### Timestamp Extraction / Watermark Emission
+
+Flink requires the events’ timestamps (each element in the stream needs to have its event timestamp assigned). This is achieved by accessing/extracting the timestamp from some field in the element. These are used to tell the system about progress in event time.
+Pravega is not aware (or does not track) of event time and does _not_ store event timestamps or watermarks.
+Nonetheless it is possible to use event time semantics via an application-specific timestamp assigner and watermark generator as described in [Flink documentation](https://ci.apache.org/projects/flink/flink-docs-release-1.3/dev/event_timestamps_watermarks.html#timestamp-assigners--watermark-generators).  
+
+Specify an `AssignerWithPeriodicWatermarks` or `AssignerWithPunctuatedWatermarks` on the `DataStream` as normal.
+
+Each parallel instance of the source processes one or more stream segments in parallel. Each watermark generator instance will receive events multiplexed from numerous segments. Be aware that segments are processed in parallel, and that no effort is made to order the events across segments in terms of their event time.  Also, a given segment may be reassigned to another parallel instance at any time, preserving exactly-once behavior but causing further spread in observed event times.
+
+
+### StreamCuts
+A `StreamCut` represents a specific position in a Pravega Stream, which may be obtained from various API interactions with the Pravega client. The `FlinkPravegaReader` accepts a `StreamCut` as the start and/or end position of a given stream. For further reading on
+StreamCuts, please refer to documentation on [StreamCut](https://github.com/pravega/pravega/blob/master/documentation/src/docs/streamcuts.md) and [sample code](https://github.com/pravega/pravega-samples/tree/v0.4.2/pravega-client-examples/src/main/java/io/pravega/example/streamcuts).
+
+#### Historical Stream Processing
+
+Historical processing refers to processing stream data from a specific position in the stream rather than from the stream's tail.  The builder API provides an overloaded method `forStream` that accepts a `StreamCut` parameter for this purpose.

```

Comment:
Could we use savepoints for historical processing as well? When would I use savepoints vs. stream cuts? 

Actually, historical stream processing means a couple of things:
1- Processing stream data at any time after it was ingested. It could be a few milliseconds after or a year after.
2- Having reproducible results independent of when the data is processed. Processing some data today must give me the same results as processing the data tomorrow or an year after.

Reply:
`savepoints` can be used to resume a (previously running) job from a checkpointed state where as `streamcuts` can be used to define the read boundary of the streams (while starting the job).